client/src/components/Q&A/AnswerModal.jsx

```diff
@@ -24,28 +54,68 @@ const AnswerModal = (props) => {
       email: emailInput,
       photos: imageInput,
     };
-    axios.post(`/qa/questions/${id}/answers`, answerInfo)
-      .then(() => {
-        toggleAnswerForm();
+    if (textCheck && nameCheck && emailCheck) {
+      axios.post(`/qa/questions/${id}/answers`, answerInfo)
+        .then(() => {
+          toggleAnswerForm();
+        })
+        .catch((error) => {
+          console.log('error in posting answer on the client', error);
+        });
+    } else {
+      alert('You must contain the following: answer, nickname, and email.');
+    }
+  };
+
+  // Get product name
+  const getProductName = () => {
+    axios.get(`/products/${currentId}`)
+      .then((response) => {
+        setProductName(response.data.name);
       })
       .catch((error) => {
-        console.log('error in posting answer on the client', error);
+        console.log('error in getting product name on the client side', error);
       });
   };
 
+  useEffect(getProductName, [currentId]);
+
   return (
     <div className="modal">
       <div className="modal-content">
         <div className="modal-header">
-          <h4 className="modal-title">Submit Answer</h4>
-        </div>
-        <div className="modal-body">
-          <textarea id="answer-entry" maxLength="200" onChange={textChangeHandler} />
-        </div>
-        <div className="modal-footer">
-          <button type="button" onClick={toggleAnswerForm}>Close</button>
-          <button type="submit" onClick={postAnswer}>Submit</button>
+          <h4 className="modal-title">Submit Your Answer</h4>
+          <span>
+            &#34;
+            {productName}
+            :&#160;
+            {question}
+            &#34;
+          </span>
         </div>
+        <form>
+          <div className="modal-body">
+            <span>Your Answer:</span>
```

Comment:
Awesome job on the Answer Modal! There seems to be a part in the business requirements that would like an asterisk next to the title where there should be a fill out. "...Each should be labelled as titled below. Those indicated as mandatory should have an asterisk next to the title."

Reply:
edited